File: src/main/java/no/nav/bidrag/beregn/forskudd/core/periode/ForskuddPeriode.java
package no.nav.bidrag.beregn.forskudd.core.periode;

import java.util.List;
import no.nav.bidrag.beregn.forskudd.core.beregning.ForskuddBeregning;
import no.nav.bidrag.beregn.forskudd.core.bo.Avvik;
import no.nav.bidrag.beregn.forskudd.core.bo.BeregnForskuddGrunnlag;
import no.nav.bidrag.beregn.forskudd.core.bo.BeregnForskuddResultat;

public interface ForskuddPeriode {

  BeregnForskuddResultat beregnPerioder(BeregnForskuddGrunnlag grunnlag);
  List<Avvik> validerInput(BeregnForskuddGrunnlag grunnlag);

  static ForskuddPeriode getInstance() {
    return new ForskuddPeriodeImpl(ForskuddBeregning.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/forskudd/core/periode/ForskuddPeriode.kt
package no.nav.bidrag.beregn.forskudd.core.periode

import no.nav.bidrag.beregn.felles.bo.Avvik
import no.nav.bidrag.beregn.forskudd.core.beregning.ForskuddBeregning
import no.nav.bidrag.beregn.forskudd.core.bo.BeregnForskuddGrunnlag
import no.nav.bidrag.beregn.forskudd.core.bo.BeregnForskuddResultat

interface ForskuddPeriode {
    fun beregnPerioder(grunnlag: BeregnForskuddGrunnlag): BeregnForskuddResultat
    fun validerInput(grunnlag: BeregnForskuddGrunnlag): List<Avvik>

    companion object {
        fun getInstance(): ForskuddPeriode {
            return ForskuddPeriodeImpl(ForskuddBeregning.getInstance())
        }
    }
}
